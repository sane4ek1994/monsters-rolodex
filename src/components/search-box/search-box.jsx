import React from 'react'

import './search-box.styles.css'

export const SearchBox = ({ className, placeholder, onChangeHandler }) => (
  <input className={className} type='search' placeholder={placeholder} onChange={onChangeHandler} />
)
