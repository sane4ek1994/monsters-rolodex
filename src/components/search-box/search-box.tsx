import { ChangeEvent } from 'react'

import './search-box.styles.css'

type SearchBoxProps = {
  className: string
  placeholder?: string
  onChangeHandler: (event: ChangeEvent<HTMLInputElement>) => void
}

export const SearchBox = ({ className, placeholder, onChangeHandler }: SearchBoxProps) => (
  <input className={className} type='search' placeholder={placeholder} onChange={onChangeHandler} />
)
