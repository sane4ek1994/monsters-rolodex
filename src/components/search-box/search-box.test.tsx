import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ChangeEvent } from 'react'

import { SearchBox } from './search-box'

it('expect to render search-box component', () => {
  const searchBoxComponent = shallow(
    <SearchBox
      className={''}
      onChangeHandler={function (event: ChangeEvent<HTMLInputElement>): void {
        throw new Error('Function not implemented.')
      }}
    />
  )

  expect(toJson(searchBoxComponent)).toMatchSnapshot()
})
