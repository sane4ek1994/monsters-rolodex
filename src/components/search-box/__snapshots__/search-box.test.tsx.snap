// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render search-box component 1`] = `
<input
  className=""
  onChange={[Function]}
  type="search"
/>
`;
